x-pack/plugins/ml/server/lib/alerts/jobs_health_service.ts

```diff
@@ -345,101 +373,126 @@ export function jobsHealthServiceProvider(
                 values: {
-                  count: softLimitCount,
-                  jobsString: softLimitJobsString,
+                  count: hardLimitCount,
+                  jobsString: hardLimitJobsString,
                 },
+              });
+            }
+
+            if (softLimitCount > 0) {
+              if (message.length > 0) {
+                message += '\n';
               }
-            );
+              message += i18n.translate(
+                'xpack.ml.alertTypes.jobsHealthAlertingRule.mmlSoftLimitMessage',
+                {
+                  defaultMessage:
+                    '{count, plural, one {Job} other {Jobs}} {jobsString} reached the soft model memory limit. Assign the job more memory or edit the datafeed filter to limit scope of analysis.',
+                  values: {
+                    count: softLimitCount,
+                    jobsString: softLimitJobsString,
+                  },
+                }
+              );
+            }
           }
 
           results.push({
+            isHealthy,
             name: HEALTH_CHECK_NAMES.mml.name,
             context: {
-              results: response,
+              results: isHealthy ? okJobs : [...(hardLimitJobs ?? []), ...(softLimitJobs ?? [])],
               message,
             },
           });
         }
       }
 
       if (config.delayedData.enabled) {
-        const response = await this.getDelayedDataReport(
-          jobs,
-          config.delayedData.timeInterval,
-          config.delayedData.docsCount
-        );
-
-        const { count, jobsString } = getJobsAlertingMessageValues(response);
+        const [exceededThresholdAnnotations, withinThresholdAnnotations] =
+          await this.getDelayedDataReport(
+            jobs,
+            config.delayedData.timeInterval,
+            config.delayedData.docsCount
+          );
 
-        if (response.length > 0) {
-          results.push({
-            name: HEALTH_CHECK_NAMES.delayedData.name,
-            context: {
-              results: response,
-              message: i18n.translate(
-                'xpack.ml.alertTypes.jobsHealthAlertingRule.delayedDataMessage',
-                {
+        const isHealthy = exceededThresholdAnnotations.length === 0;
+        const { count, jobsString } = getJobsAlertingMessageValues(exceededThresholdAnnotations);
+
+        results.push({
+          isHealthy,
+          name: HEALTH_CHECK_NAMES.delayedData.name,
+          context: {
+            results: isHealthy ? withinThresholdAnnotations : exceededThresholdAnnotations,
+            message: isHealthy
+              ? i18n.translate(
+                  'xpack.ml.alertTypes.jobsHealthAlertingRule.delayedDataRecoveryMessage',
+                  {
+                    defaultMessage: 'No jobs are suffering from delayed data.',
```

Comment:
```suggestion
                    defaultMessage: 'No data delay has occurred.',
```

Reply:
Updated in https://github.com/elastic/kibana/pull/132496/commits/f69b1240e7dfe172697f7af8c029c695a6fb1dc9